File: LLM-Translations/GPT-4/codenet/Python/Go/s538657784.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var N int
	fmt.Scan(&N)
	sqN := int(math.Sqrt(float64(N)))
	for i := 0; i < sqN; i++ {
		if N%(sqN-i) == 0 {
			fmt.Println(int(N/(sqN-i)+sqN-i) - 2)
			break
		}
	}
}
// 

File: LLM-Translations/GPT-4/codenet/Python/C/s538657784.c

#include <stdio.h>
#include <math.h>

int main() {
    int N, sqN;
    scanf("%d", &N);
    sqN = (int)sqrt(N);
    for (int i = 0; i < sqN; i++) {
        if (N % (sqN - i) == 0) {
            printf("%d\n", (int)(N / (sqN - i) + sqN - i) - 2);
            break;
        }
    }
    return 0;
}


